contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -190,122 +199,68 @@ contract SoteriaCoverageProduct is ISoteriaCoverageProduct, ERC721, EIP712, Reen
     }
 
     /**
-     * @notice Charge premiums for each policy holder.
-     * @param holders_ The policy holders.
-     * @param premiums_ The premium amounts in `wei` per policy holder.
-    */
-    function chargePremiums(address[] calldata holders_, uint256[] calldata premiums_) external payable override onlyGovernance whileUnpaused {
-        uint256 count = holders_.length;
-        require(count == premiums_.length, "length mismatch");
-        require(count <= policyCount(), "policy count exceeded");
-        uint256 amountToPayTreasury = 0;
-
-        for (uint256 i = 0; i < count; i++) {
-            // skip computation if policy inactive (coverAmount == 0)
-            if (coverAmountOf(policyByOwner(holders_[i])) == 0) {
-                continue;
-            }
-
-            if (_funds[holders_[i]] >= premiums_[i]) {
-                amountToPayTreasury += premiums_[i];
-                _funds[holders_[i]] -= premiums_[i];
-                emit PremiumCharged(holders_[i], premiums_[i]);
-            } else {
-                uint256 partialPremium = _funds[holders_[i]];
-                amountToPayTreasury += partialPremium;
-                _funds[holders_[i]] = 0;
-                // turn off the policy
-                _closePolicy(holders_[i]);
-                emit PremiumPartiallyCharged(holders_[i], premiums_[i], partialPremium);
-            }  
-        }
-        // transfer premium to the treasury
-        ITreasury(payable(_registry.treasury())).routePremiums{value: amountToPayTreasury}();
+     * @notice Withdraw ETH from Soteria account to user.
+     * @param amount_ Amount policyholder desires to withdraw.
+     * User Soteria account must have > minAccountBalance.
+     * Otherwise account will be deactivated.
+     */
+    function withdraw(uint256 amount_) external override nonReentrant whileUnpaused {
+      require(amount_ <= _accountBalanceOf[msg.sender], "cannot withdraw this amount");
+      
+      uint256 currentCoverLimit = _coverLimitOf[_policyOf[msg.sender]];
+
+      if (_accountBalanceOf[msg.sender] - amount_ > ( _maxRateNum * _chargeCycle * currentCoverLimit ) / _maxRateDenom) {
+          _withdraw(msg.sender, amount_);
+      } else {
+          uint256 accountBalance = _accountBalanceOf[msg.sender];
```

Comment:
The redundant **uint256 accountBalance = _accountBalanceOf[msg.sender];** statement. Just call _accountBalanceOf[msg.sender] in the transfer.

Reply:
Need to store the original value of `_accountBalanceOf[msg.sender]` in the placeholder `uint256 accountBalance`, because the next line `_deactivatePolicy(policyOwner);` sets `_accountBalanceOf[msg.sender] = 0`